packages/rocketchat-api/server/v1/users.js

```diff
@@ -220,24 +220,42 @@ RocketChat.API.v1.addRoute('users.setAvatar', { authRequired: true }, {
 				RocketChat.setUserAvatar(user, this.bodyParams.avatarUrl, '', 'url');
 			} else {
 				const busboy = new Busboy({ headers: this.request.headers });
+				let fields = {};
+				const getUserFromFormData = (fields) => {
+					let userFromFormData;
+					if (fields.userId) {
+						userFromFormData = RocketChat.models.Users.findOneById(fields.userId);
+					} else if (fields.username) {
+						userFromFormData = RocketChat.models.Users.findOneByUsername(fields.username);
+					}
+					return userFromFormData;
+				};
 
 				Meteor.wrapAsync((callback) => {
 					busboy.on('file', Meteor.bindEnvironment((fieldname, file, filename, encoding, mimetype) => {
 						if (fieldname !== 'image') {
 							return callback(new Meteor.Error('invalid-field'));
 						}
-
 						const imageData = [];
 						file.on('data', Meteor.bindEnvironment((data) => {
 							imageData.push(data);
 						}));
 
 						file.on('end', Meteor.bindEnvironment(() => {
+							const sentTheUserByFormData = fields.userId || fields.username;
+							if (sentTheUserByFormData) {
+								user = getUserFromFormData(fields);
```

Comment:
Where are we checking the permissions to change the other user's avatar?

Reply:
at [line 250](https://github.com/RocketChat/Rocket.Chat/pull/11431/files#diff-1d5c7afdaba4728789f33cfa02ec1515R250)